src/resources/filters/quarto-pre/line-numbers.lua

```diff
@@ -1,29 +1,35 @@
 -- line-numbers.lua
 -- Copyright (C) 2020-2022 Posit Software, PBC
 
+local constants = require("modules/constants")
 
 function line_numbers()
   return {
     CodeBlock = function(el)
       if #el.attr.classes > 0 then
         local lineNumbers = lineNumbersAttribute(el)
+        el.attr.attributes[constants.kCodeLineNumbers] = nil
         if lineNumbers ~= false then
           -- use the pandoc line numbering class
           el.attr.classes:insert("number-lines")
-          -- remove for all formats except reveal
-          if not _quarto.format.isRevealJsOutput() then
-            el.attr.attributes["code-line-numbers"] = nil
+          -- remove for all formats except reveal and docusaurus
+          if type(lineNumbers) == "string" and (_quarto.format.isRevealJsOutput() or _quarto.format.isDocusaurusOutput()) then
```

Comment:
It seems to me that this would mean that the new formats added by extension would not be able to use this feature unless the Quarto CLI source code is updated, which would not be ideal, would it?

Reply:
Sorry by I don't understand the question. The docusaurus "extension" is built in now: 
- https://quarto.org/docs/output-formats/docusaurus.html
- https://github.com/quarto-dev/quarto-cli/tree/main/src/resources/extensions/quarto/docusaurus

So any change to it will require a new update of Quarto CLI. 

Also `code-line-numbers` attributes is handled in Quarto source code so this will require a new version of the CLI. 

What do you have in mind ?

Reply:
Sorry my question is not clear.
For example, it seems that Material for MkDocs has a similar feature, but if someone has created a Quarto extension for Material for MkDocs, they cannot access here.
https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#highlighting-specific-lines

Reply:
Ok I understand. 

> if someone has created a Quarto extension for Material for MkDocs (off course outside of quarto-cli)

Is this hypothetical or is there one which needs this ? Just to gauge priority here. 

It seems to me this is a new feature where a format would need to declare it is supporting line highlighting. We choose to support `code-line-numbers` options for line highlight too, but this means that we need to teach quarto which format can support it, as we don't have a mechanism yet for formats to tells quarto. 

@dragonstyle do we have already some mechanism for an extension format (or a format in general) to tell some quarto process that something is supported ? 

In this case we need to know when to pass `code-line-numbers: <string>` as attributes on the CodeCell for the format to process. This is not a Pandoc feature so it requires dedicated processing

* Revealjs handles it in a Plugin
* Docusaurus handles it in the Lua Writer
* a mkDocs extension would need to do the same. 

Anyhow, if we were to support that, this would require a new CLI version anyway. I'll probably merge this and we'll see how to support more formats in the future

Reply:
I opened https://github.com/quarto-dev/quarto-cli/issues/7213

Reply:
@cderv Thank you for summarizing this. I agree that merging this is sufficient for now.

> Is this hypothetical or is there one which needs this ? Just to gauge priority here.

I just wanted to point out that this is an issue with the current implementation.
I personally don't want to use Quarto and Material for MkDocs in combination right away (I currently prefer Docusaurus!).

However, given the popularity of Material for Mkdocs, I'm sure there will be users who need something like that.

Reply:
Ok good. I'll take note on that and see if I can add that for 1.4 maybe. Seems easy enough.